src/test/java/org/aksw/iguana/cc/query/QueryDataTest.java

```diff
@@ -0,0 +1,65 @@
+package org.aksw.iguana.cc.query;
+
+import org.aksw.iguana.cc.query.source.QuerySource;
+import org.aksw.iguana.cc.query.source.impl.FileSeparatorQuerySource;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class QueryDataTest {
+
+    private static Path tempFile = null;
+
+    @BeforeAll
+    public static void setup() throws IOException {
+        tempFile = Files.createTempFile("test", "txt");
+        Files.writeString(tempFile, """
+                SELECT ?s ?p ?o WHERE {
+                    ?s ?p ?o
+                }
+                
+                INSERT DATA {
```

Comment:
Just realized we are also missing something here. Please create an issue to also support multiple semicolon-chained update operations. 

https://www.w3.org/TR/2013/REC-sparql11-update-20130321/#updateLanguage

Reply:
What exactly is the problem with semicolon-chained update operations? Should they be split up into multiple queries?

Reply:
Good question. They should just work, right? Only, We should probably double check that LSQ handles them correctly when we include that.